lib/commands/memory.js

```diff
@@ -45,7 +46,7 @@ export default {
       );
     }
     this.log.info(`Emulating Low Memory warning for the process id ${pids[0]}, bundle id ${bundleId}`);
-    await devicectl.sendMemoryWarning(pids[0]);
+    await device.devicectl.sendMemoryWarning(pids[0]);
```

Comment:
perhaps we need to disable json output option for this method since we anyway not use it

Reply:
the command itself failed by the devicectl, so the json output itself had no issue here (for the failure in the description)
```
% xcrun devicectl device process sendMemoryWarning --device 00008020-000E5CDA0A23002E --pid 4607
01:51:41  Enabling developer disk image services.
01:51:41  Acquired usage assertion.
ERROR: The operation couldn’t be completed. No such file or directory (NSPOSIXErrorDomain error 2.)
```

Reply:
maybe this cannot be done for system apps

Reply:
The same error occurred in netflix and my simple app package as well... I actually tested similar apps in https://github.com/appium/appium-xcuitest-driver/pull/2308 , so  I guess it has more conditions to make it work on the app side... although this command itself should be ok...

Reply:
hm, the device itself should have the memory warning capability:
Let me check further if it has more condition to make it work. anyway this is not Appium side's issue

```
% xcrun devicectl device  info details --device 00008020-000E5CDA0A23002E
Gathering device information...
Current device information:
• identifier: D7065391-DA0E-4CED-BE17-3A6616B5A466
▿ hardwareProperties:
    • cpuType: arm64e (type: 16777228, subtype: 2)
    • deviceType: iPad
    • ecid: 4042741216641070
    • hardwareModel: J210AP
    • internalStorageCapacity: 64000000000
    • isProductionFused: true
    • marketingName: iPad mini (5th generation)
    • platform: iOS
    • productType: iPad11,1
    • reality: physical
    • serialNumber: DMPZP204LM93
    ▿ supportedCPUTypes:
        • arm64e (type: 16777228, subtype: 2)
        • arm64 (type: 16777228, subtype: 0)
        • arm64 (type: 16777228, subtype: 1)
        • arm64_32 (type: 33554444, subtype: 1)
    • supportedDeviceFamilies: [1, 2]
    • thinningProductType: iPad11,1
    • udid: 00008020-000E5CDA0A23002E
▿ deviceProperties:
    • bootState: booted
    • ddiServicesAvailable: true
    • developerModeStatus: enabled
    • hasInternalOSBuild: false
    • name: iPad
    • osBuildUpdate: 21E217
    • osVersionNumber: 17.4
    • rootFileSystemIsWritable: false
    • screenViewingURL: coredevice-devices:/viewDeviceByUUID?uuid=D7065391-DA0E-4CED-BE17-3A6616B5A466
    • bootedFromSnapshot: true
    • bootedSnapshotName: com.apple.os.update-D3867D58340A072FEBB096505E4659DBF974A51A6626D34F458944991A273D18
▿ connectionProperties:
    • authenticationType: manualPairing
    • isMobileDeviceOnly: false
    • lastConnectionDate: 2024-03-24 16:35:48 +0000
    ▿ localHostnames:
        • iPad.coredevice.local: iPad.coredevice.local
        • 00008020-000E5CDA0A23002E.coredevice.local: 00008020-000E5CDA0A23002E.coredevice.local
        • D7065391-DA0E-4CED-BE17-3A6616B5A466.coredevice.local: D7065391-DA0E-4CED-BE17-3A6616B5A466.coredevice.local
    ▿ potentialHostnames:
        • 00008020-000E5CDA0A23002E.coredevice.local: 00008020-000E5CDA0A23002E.coredevice.local
        • D7065391-DA0E-4CED-BE17-3A6616B5A466.coredevice.local: D7065391-DA0E-4CED-BE17-3A6616B5A466.coredevice.local
    • pairingState: paired
    • transportType: wired
    • tunnelTransportProtocol: tcp
    • tunnelIPAddress: fded:c54:ad8a::1
    • tunnelState: connected
▿ capabilities:
    • Transfer Files (com.apple.coredevice.feature.transferFiles)
    • List Applications (com.apple.coredevice.feature.listapps)
    • com.apple.instruments.dtservicehub (com.apple.dt.profile)
    • Spawn Executable (com.apple.coredevice.feature.spawnexecutable)
    • Reboot Device (com.apple.coredevice.feature.rebootdevice)
    • Monitor Process for Termination (com.apple.coredevice.feature.monitorprocesstermination)
    • com.apple.security.cryptexd.remote (Cryptex1)
    • Query MobileGestalt (com.apple.coredevice.feature.querymobilegestalt)
    • Send Memory Warning to Process (com.apple.coredevice.feature.sendmemorywarningtoprocess)
    • com.apple.security.cryptexd.remote (ReadIdentifiers)
    • com.apple.dt.remoteFetchSymbols (com.apple.dt.remoteFetchSymbols.dyldSharedCacheFiles)
    • View Device Screen (com.apple.coredevice.feature.viewdevicescreen)
    • Send Signal to Process (com.apple.coredevice.feature.sendsignaltoprocess)
    • Get Display Information (com.apple.coredevice.feature.getdisplayinfo)
    • Rsync Files (com.apple.coredevice.feature.rsyncfiles)
    • Get Lock State (com.apple.coredevice.feature.getlockstate)
    • com.apple.security.cryptexd.remote (CryptexInstall)
    • Install Root (com.apple.coredevice.feature.installroot)
    • List Processes (com.apple.coredevice.feature.listprocesses)
    • Fetch Developer Disk Image Services Metadata (com.apple.coredevice.feature.fetchddimetadata)
    • Disconnect from Device (com.apple.coredevice.feature.disconnectdevice)
    • Launch Application (com.apple.coredevice.feature.launchapplication)
    • Install Application (com.apple.coredevice.feature.installapp)
    • Uninstall Root (com.apple.coredevice.feature.uninstallroot)
    • com.apple.internal.dt.remote.debugproxy (com.apple.coredevice.feature.debugserverproxy)
    • Capture Sysdiagnose (com.apple.coredevice.feature.capturesysdiagnose)
    • Acquire Usage Assertion (com.apple.coredevice.feature.acquireusageassertion)
    • Uninstall Application (com.apple.coredevice.feature.uninstallapp)
    • Fetch Application Icons (com.apple.coredevice.feature.fetchappicons)
    • List Roots (com.apple.coredevice.feature.listroots)
    • Fetch Extended Device Info (com.apple.coredevice.feature.getdeviceinfo)
    • Disable Developer Disk Image Services (com.apple.coredevice.feature.disableddiservices)
    • Unpair Device (com.apple.coredevice.feature.unpairdevice)
• visibilityClass: default
```